modules/xfem/include/userobjects/GeometricCutUserObject.h

```diff
@@ -19,27 +19,69 @@
 
 using namespace libMesh;
 
+class XFEM;
+
+namespace Xfem
+{
+/// Data structure defining a cut on an element edge
 struct CutEdge
 {
+  /// ID of the first node on the edge
   unsigned int id1;
+  /// ID of the second node on the edge
   unsigned int id2;
+  /// Fractional distance along the edge (from node 1 to 2) where the cut is located
   Real distance;
+  /// Local ID of this side in the host element
   unsigned int host_side_id;
 };
 
+/// Data structure defining a cut through a node
 struct CutNode
 {
+  /// ID of the cut node
   unsigned int id;
+  /// Local ID of this node in the host element
   unsigned int host_id;
 };
 
+/// Data structure defining a cut through a face
 struct CutFace
 {
+  /// ID of the cut face
   unsigned int face_id;
+  /// IDs of all cut faces
   std::vector<unsigned int> face_edge;
+  /// Fractional distance along the cut edges where the cut is located
   std::vector<Real> position;
 };
 
+/// Data structure describing geometrically described cut through 2D element
+struct GeomMarkedElemInfo2D
+{
+  /// Container for data about all cut edges in this element
+  std::vector<CutEdge> elem_cut_edges;
+  /// Container for data about all cut nodes in this element
+  std::vector<CutNode> elem_cut_nodes;
+  /// Container for data about all cut fragments in this element
+  std::vector<CutEdge> frag_cut_edges;
+  /// Container for data about all cut edges in cut fragments in this element
+  std::vector<std::vector<Point>> frag_edges;
+};
```

Comment:
This has come up before, but we've decided that even for structs, that we continue to use the same naming convention that we do for classes (e.g. leading underscore for members).